### PR TITLE
Register Retry, SymDict and XMLDict

### PR DESCRIPTION
**Pull request by samoconnor**
https://github.com/samoconnor/Retry.jl
https://github.com/samoconnor/SymDict.jl
https://github.com/samoconnor/XMLDict.jl (was https://github.com/JuliaLang/LightXML.jl/pull/39)

### CODE REVIEW

File: XMLDict/versions/0.0.1/requires [hunk 1]
```diff
@@ -0,0 +1,2 @@
+julia 0.4
+DataStructures
```
Comment (tkelman) [inline XMLDict/versions/0.0.1/requires]: missing dependency on lightxml? do turn travis on for these

**Comment by tkelman** [comment]
XMLDict is defining Base methods on LightXML types, which means any user of LightXML would have their code's behavior change by using XMLDict. This is a bit sketchy since it could be some deep dependency of some other package and it would still change LightXML behaviour elsewhere in a project.

**Comment by samoconnor** [comment]
Wrapper type added in https://github.com/samoconnor/XMLDict.jl/commit/0ccc6a691774898ec45d7382543ddf2ab33c7822

**Comment by samoconnor** [comment]
https://github.com/samoconnor/XMLDict.jl/commit/d13f6d77678f8dd0504f56e9832e625a1646ea3e makes the wrapper type `<: Associative` and cleans up `get()` vs `getindex()`, `haskey()` etc.

The only things now exported are:
 - `Base.get(::XMLDictElement,...)`
 - `parse_xml(l::AbstractString)`
 - `xml_dict(::Union{AbstractString, XMLDocument},...)`

Are there any other issues ?

**Comment by tkelman** [comment]
master of XMLDict looks fine now and okay to register.

The current tags are missing the LightXML dependency, and have nonconforming url's - see the Travis failure here. Would also be best practices to enable Travis for all packages that don't have particular requirements preventing them from passing tests there.

**Comment by samoconnor** [comment]
I think the URLs are all fixed now and I've pushed a new tag for XMLDIct.
I will try to get travis going, but I'd rather that didn't hold up this PR.

**Comment by tkelman** [comment]
"Retry" is on the generic side for a package name, but if no one objects much to that then I think this should be okay now if it passes the metadata travis check.

**Comment by samoconnor** [comment]
Travis results:
* https://travis-ci.org/samoconnor/Retry.jl
* https://travis-ci.org/samoconnor/SymDict.jl
* https://travis-ci.org/samoconnor/FNVHash.jl

I haven't enabled XMLDict yet, it relies on shelling out to `xmllint` i'm not sure if that will work, but I'll try it now...

**Comment by tkelman** [comment]
If it doesn't work right away, try adding
```
addons:
  apt:
    packages:
      - libxml2-utils
```
to `.travis.yml`

**Comment by samoconnor** [comment]
XMLDict does `using JSON` in `runtests.jl` but `XMLDict.jl` does not require `JSON`.
What's the best way to tell travis to install JSON.jl?

**Comment by tkelman** [comment]
Add `JSON` to `test/REQUIRE`

**Comment by samoconnor** [comment]
you make it sound easy :)

**Comment by tkelman** [comment]
I think this stuff is written down in the docs (ours and/or travis') but if it needs clarification, suggestions welcome.

**Comment by samoconnor** [comment]
The XMLDict.jl build failed because of this upstream bug: https://github.com/JuliaLang/LightXML.jl/issues/40

adding a work-around now...

**Comment by samoconnor** [comment]
https://travis-ci.org/samoconnor/XMLDict.jl running ok now

**Comment by samoconnor** [comment]
Hi @tkelman, I've figured out how to [embed test AWS credentials into the travis config](https://github.com/samoconnor/AWSCore.jl/blob/master/.travis.yml) to get my AWS modules working on travis.
The [secure env vars seem to work ok](https://travis-ci.org/samoconnor/AWSCore.jl/jobs/100765965) but for now it won't actually run because it depends on Retry, SymDict and XMLDict.

It would be great if you could merge this PR so I can keep going...

**Comment by tkelman** [comment]
By

> "Retry" is on the generic side for a package name, but if no one objects much to that then I think this should be okay now if it passes the metadata travis check.

I meant "I'll be leaving this open for comments for a little longer." It's been open for 4 days but only in a merge-able state for a few hours. Good enough I suppose.

**Comment by samoconnor** [comment]
thx @tkelman for all your feedback and helping me figure out the doc and testing infrastructure.

**Comment by nalimilan** [comment]
My two cents on the name Retry.jl: indeed that's quite general, but I don't see any other obvious meaning of this verb, so that's probably fine.

**Pull request merged**